### PR TITLE
Various client improvements

### PR DESCRIPTION
**Pull request by tboerger**
* Syncronized hound and codeclimate configs with other crowbar projects
* Use travis containers
* Removed todo file and added issues instead
* Added formatter and filter classes
* Rebuilt role commands based on formatter and filter

### CODE REVIEW

File: lib/crowbar/client/command/role.rb [hunk 1]
```diff
@@ -65,16 +80,19 @@ module Role
                 Request.instance.role_show(barclamp, role) do |request|
                   case request.code
                   when 200
-                    body = begin
-                      JSON.parse(request.body)
-                    rescue
-                      []
-                    end
+                    formatter = Formatter::Array.new(
```
Comment (MaximilianMeister) [inline lib/crowbar/client/command/role.rb]: MINOR: `formatter` looks a bit DRY. can you put that into a method with a headings parameter?

**Comment by MaximilianMeister** [comment]
other than the DRY it looks good to me

File: lib/crowbar/client/command/role.rb (hunk 1 above)
Comment (tboerger) [inline lib/crowbar/client/command/role.rb]: Uhm, formatter got always different parameters like opts from the current context, specific headings and request from the current context.

Comment (tboerger) [inline lib/crowbar/client/command/role.rb]: And in general that's not all I want to refactor and make more abstract, just the beginning :)
I will add some tests as well when I get back to this.

**Comment by tboerger** [comment]
Merging it, it's anyway not used yet and under strong development.

**Pull request merged**